drivers/ethernet/eth_mcux.c

```diff
@@ -1238,299 +1236,167 @@ static void eth_mcux_error_isr(const struct device *dev)
+	COND_CODE_1(DT_INST_IRQ_HAS_NAME(n, name),			\
+		    (ETH_MCUX_IRQ_INIT(n, name)),			\
+		    (ETH_MCUX_NONE))
 
 #if defined(CONFIG_PTP_CLOCK_MCUX)
-#if DT_NODE_HAS_STATUS(PTP_INST_NODEID(0), okay)
-	IRQ_CONNECT(DT_IRQ_BY_NAME(PTP_INST_NODEID(0), ieee1588_tmr, irq),
-		    DT_IRQ_BY_NAME(PTP_INST_NODEID(0), ieee1588_tmr, priority),
-		    eth_mcux_ptp_isr,
-		    DEVICE_GET(eth_mcux_0),
-		    0);
-	irq_enable(DT_IRQ_BY_NAME(PTP_INST_NODEID(0), ieee1588_tmr, irq));
-#endif
-#endif
-}
-#endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay) */
-
-#if DT_NODE_HAS_STATUS(DT_DRV_INST(1), okay)
-
-#if DT_INST_PROP(1, zephyr_random_mac_address) && \
-	NODE_HAS_VALID_MAC_ADDR(DT_DRV_INST(1))
-#error Conflict between 'local-mac-address' and 'zephyr,random-mac-address'
-#endif
+#define PTP_INST_NODEID(n) DT_CHILD(DT_DRV_INST(n), ptp)
 
-#if !NODE_HAS_VALID_MAC_ADDR(DT_DRV_INST(1))
-static void generate_eth1_mac(uint8_t *mac_addr)
-{
-#if DT_INST_PROP(1, zephyr_random_mac_address)
-	gen_random_mac(mac_addr,
-		       FREESCALE_OUI_B0, FREESCALE_OUI_B1, FREESCALE_OUI_B2);
+#define ETH_MCUX_IRQ_PTP_INIT(n)							\
+	do {										\
+		IRQ_CONNECT(DT_IRQ_BY_NAME(PTP_INST_NODEID(n), ieee1588_tmr, irq),	\
+			    DT_IRQ_BY_NAME(PTP_INST_NODEID(n), ieee1588_tmr, priority),	\
+			    eth_mcux_ptp_isr,						\
+			    DEVICE_GET(eth_mcux_##n),					\
+			    0);								\
+		irq_enable(DT_IRQ_BY_NAME(PTP_INST_NODEID(n), ieee1588_tmr, irq));	\
+	} while (0)
+
+#define ETH_MCUX_IRQ_PTP(n)						\
+	COND_CODE_1(DT_NODE_HAS_STATUS(PTP_INST_NODEID(n), okay),	\
+		    (ETH_MCUX_IRQ_PTP_INIT(n)),				\
+		    (ETH_MCUX_NONE))
 #else
-	/* Generate_unique_mac */
-	mac_addr[0] = FREESCALE_OUI_B0;
-	mac_addr[1] = FREESCALE_OUI_B1;
-	mac_addr[2] = FREESCALE_OUI_B2;
-	/* Trivially "hash" up to 128 bits of MCU unique identifier */
-#if defined(CONFIG_SOC_SERIES_IMX_RT)
-	uint32_t id = OCOTP->CFG1 ^ OCOTP->CFG2;
-#elif defined(CONFIG_SOC_SERIES_KINETIS_K6X)
-	uint32_t id = SIM->UIDH ^ SIM->UIDMH ^ SIM->UIDML ^ SIM->UIDL;
-#else
-#error "Unsupported SOC"
+#define ETH_MCUX_IRQ_PTP(n)
 #endif
-	mac_addr[0] |= 0x02; /* force LAA bit */
 
-	mac_addr[3] = id >> 8;
-	mac_addr[4] = id >> 16;
-	mac_addr[5] = id >> 0;
+#define ETH_MCUX_GENERATE_MAC_RANDOM(n)					\
+	static void generate_eth##n##_mac(uint8_t *mac_addr)		\
+	{								\
+		gen_random_mac(mac_addr,				\
+			       FREESCALE_OUI_B0,			\
+			       FREESCALE_OUI_B1,			\
+			       FREESCALE_OUI_B2);			\
+	}
 
-	mac_addr[5] += 1;
-#endif /* zephyr_random_mac_address */
-}
-#endif
+#define ETH_MCUX_GENERATE_MAC_UNIQUE(n)					\
+	static void generate_eth##n##_mac(uint8_t *mac_addr)		\
+	{								\
+		mac_addr[0] = FREESCALE_OUI_B0;				\
+		mac_addr[1] = FREESCALE_OUI_B1;				\
+		mac_addr[2] = FREESCALE_OUI_B2;				\
+		uint32_t id = ETH_MCUX_ID;				\
+		mac_addr[0] |= 0x02; /* force LAA bit */		\
+		mac_addr[3] = id >> 8;					\
+		mac_addr[4] = id >> 16;					\
+		mac_addr[5] = id >> 0;					\
+		mac_addr[5] += n;					\
```

Comment:
This certainly works, but imho is hard to read. Could we change this to
```
		uint32_t id = ETH_MCUX_ID;				\
		                                               		\
		mac_addr[0] = FREESCALE_OUI_B0;				\
		mac_addr[0] |= 0x02; /* force LAA bit */		\
		mac_addr[1] = FREESCALE_OUI_B1;				\
		mac_addr[2] = FREESCALE_OUI_B2;				\
		mac_addr[3] = id >> 8;					\
		mac_addr[4] = id >> 16;					\
		mac_addr[5] = id >> 0;					\
		mac_addr[5] += n;		
```

Reply:
Yes, done